runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -249,6 +254,64 @@ private String decorateTarget(String oldTarget, Decorations decorations)
 		return newTarget;
 	}
 
+	/*
+		Changes the player name for the trade window title to be green if they are a friend.
+	 */
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired event)
+	{
+		int scriptId = event.getScriptId();
+
+		if (scriptId == ScriptID.TRADE_MAIN_INIT)
+		{
+			clientThread.invokeLater(() ->
+			{
+				Widget tradeTitle = client.getWidget(WidgetInfo.TRADE_WINDOW_HEADER);
+				String header = tradeTitle.getText();
+				String playerName = header.substring("Trading with: ".length());
+				header = header.substring(0, "Trading with: ".length());
+
+				Player[] players = client.getCachedPlayers();
```

Comment:
You can use `client.getPlayers()` instead to remove the need for null-checking since the list it returns is not sparse, unlike `getCachedPlayers()`.